overlord/devicestate/users.go

```diff
@@ -185,8 +190,10 @@ func createKnownSystemUser(state *state.State, userAssertion *asserts.SystemUser
 	// the assertion against the current brand/model/time
 	username, expiration, addUserOpts, err := getUserDetailsFromAssertion(assertDb, model, serial, email)
 	if err != nil {
-		if err == errSystemUserBoundToSerialButTooEarly {
-			// TODO retry later once we have acquired a device serial
+		if errors.Is(err, errSystemUserBoundToSerialButTooEarly) {
+			state.Set("assertion-waiting-on-serial", true)
+			logger.Noticef("waiting for serial to add user %q: %s", email, err)
+			return nil, nil
 		}
 		logger.Noticef("ignoring system-user assertion for %q: %s", email, err)
```

Comment:
this is fine here, though there is an argument to be made that this check should be moved one level up into createAllKnownSystemUsers in case createKnownSystemUser ends up being reused in a different context